Sources/ClientRuntime/Networking/Http/HttpClientConfiguration.swift

```diff
@@ -1,17 +1,45 @@
-/*
- * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
- * SPDX-License-Identifier: Apache-2.0.
- */
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.TimeInterval
 
 public class HttpClientConfiguration {
-    public var protocolType: ProtocolType
-    // initialize with default headers
+
+    /// The timeout for a request, in seconds.
+    ///
+    /// If none is provided, the client will use default values based on the platform.
+    public var connectTimeout: TimeInterval?
+
+    /// HTTP headers to be submitted with every HTTP request.
+    ///
+    /// If none is provided, defaults to no extra headers.
     public var defaultHeaders: Headers
+
     // add any other properties here you want to give the service operations
     // control over to be mapped to the Http Client
 
-    public init(protocolType: ProtocolType = .https,
-                defaultHeaders: Headers = Headers()) {
+    /// The URL scheme to be used for HTTP requests.  Supported values are `http` and `https`.
+    ///
+    /// If none is provided, the default protocol for the operation will be used
+    public var protocolType: ProtocolType?
+
+    /// Creates a configuration object for a SDK HTTP client.
+    ///
+    /// Not all configuration settings may be followed by all clients.
+    /// - Parameters:
+    ///   - connectTimeout: The maximum time to wait for a response without receiving any data.
+    ///   - defaultHeaders: HTTP headers to be included with every HTTP request.
+    ///   Note that certain headers may cause your API request to fail.  Defaults to no headers.
+    ///   - protocolType: The HTTP scheme (`http` or `https`) to be used for API requests.  Defaults to the operation's standard configuration.
+    public init(
+        connectTimeout: TimeInterval? = nil,
+        defaultHeaders: Headers = Headers(),
+        protocolType: ProtocolType? = nil
+    ) {
```

Comment:
I will change this param back to `protocolType: ProtocolType = .https` to avoid breaking the existing API.
I also added a line to actually use the `connectTimeout` param.